source/templates.tex

```diff
@@ -7260,32 +7261,31 @@
 \end{codeblock}
 \end{example}
 
-When all template arguments have been deduced or obtained from
-default template arguments, all uses of template parameters in
-the template parameter list of the template and the function type
-are replaced with the corresponding deduced
-or default argument values. If the substitution results in an
-invalid type, as described above, type deduction fails.
+Both the deduced and default template argument values are substituted
+for their corresponding template parameters. If the substitution results
+in an invalid type, as described above, type deduction fails.
 If the function template has associated constraints\iref{temp.constr.decl},
 those constraints are checked for satisfaction\iref{temp.constr.constr}.
 If the constraints are not satisfied, type deduction fails.
 
 \pnum
 At certain points in the template argument deduction process it is necessary
-to take a function type that makes use of template parameters and replace those
-template parameters with the corresponding template arguments. This is done at
-the beginning of template argument deduction when any explicitly specified
-template arguments are substituted into the function type, and again at the end
-of template argument deduction when any template arguments that were deduced or
-obtained from default arguments are substituted.
+to take a function type that makes use of template parameters and substitute
+those template parameters with the corresponding template arguments.
+Substitution is done at the beginning of template argument deduction when
+any explicitly specified template arguments are substituted into the
+function type, and again at the end of template argument deduction when
+any template arguments that were deduced or obtained from default
+arguments are substituted.
 
 \pnum
-The substitution occurs in all types and expressions that are used in the function
+Substitution of template argument values with the corresponding template
+parameters occurs in all types and expressions that are used in the function
```

Comment:
This sounds backwards: we should substitute the parameters with the arguments, not vice versa.

Reply:
You are absolutely correct. I should have written for, instead of with. Does this sound okay:

> Substitution of template argument values for the corresponding template
parameters occurs in all types and expressions that are used in the function